reactive/kotlinx-coroutines-reactor/src/Mono.kt

```diff
@@ -34,6 +33,50 @@ public fun <T> mono(
     return monoInternal(GlobalScope, context, block)
 }
 
+/**
+ * Awaits the single value from the given [Mono] without blocking the thread and returns the resulting value, or, if
+ * this publisher has produced an error, throws the corresponding exception. If the Mono completed without a value,
+ * `null` is returned.
+ *
+ * This suspending function is cancellable.
+ * If the [Job] of the current coroutine is cancelled or completed while the suspending function is waiting, this
+ * function immediately cancels its [Subscription] and resumes with [CancellationException].
+ */
+public suspend fun <T> Mono<T>.awaitSingleOrNull(): T? = suspendCancellableCoroutine { cont ->
+    injectCoroutineContext(cont.context).subscribe(object : Subscriber<T> {
+        private var seenValue = false
+
+        override fun onSubscribe(s: Subscription) {
+            cont.invokeOnCancellation { s.cancel() }
+            s.request(Long.MAX_VALUE)
+        }
+
+        override fun onComplete() {
+            if (!seenValue)
+                cont.resume(null)
```

Comment:
[nit] I'd suggest sticking to either one-liners or using explicit scoping with `{}`